docs/source/graph-construction/gs-processing/gspartition/sagemaker.rst

```diff
@@ -0,0 +1,134 @@
+===================================
+Running partition jobs on SageMaker
+===================================
+
+Once the :ref:`distributed processing setup<gsprocessing_distributed_setup>` is complete,
+you can use Amazon SageMaker launch scripts to launch distributed processing jobs with AWS resources.
+
+Build the docker image for partition jobs on SageMaker
+------------------------------------------------------
+GSPartition job on SageMaker uses SageMaker's **BYOC** (Bring Your Own Container) mode.
+
+Step 1: Build a SageMaker-compatible Docker image
+...................................................
+
+.. note::
+    * Please make sure your account has access key (AK) and security access key (SK) configured to authenticate accesses to AWS services, users can refer to `example policy <https://docs.aws.amazon.com/AmazonECR/latest/userguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-access-one-bucket>`_.
+    * For more details of Amazon ECR operation via CLI, users can refer to the `Using Amazon ECR with the AWS CLI document <https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-cli.html>`_.
+
+First, in a Linux machine, configure a Docker environment by following the `Docker documentation <https://docs.docker.com/get-docker/>`_ suggestions.
+
+In order to use the SageMaker base Docker image, users need to refer the `DLC image command <https://github.com/aws/deep-learning-containers/blob/master/available_images.md>`_
+to retrieve the specific docker image command. For example:
+
+.. code-block:: bash
+
+    aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 763104351884.dkr.ecr.us-east-1.amazonaws.com
+
+.. note::
+    For region other than ``us-east-1``, please refer to `available region <https://docs.aws.amazon.com/sagemaker/latest/dg-ecr-paths/sagemaker-algo-docker-registry-paths.html>`_
+
+Then, clone GraphStorm source code, and build a GraphStorm SageMaker compatible Docker image from source with commands:
+
+.. code-block:: bash
+
+    git clone https://github.com/awslabs/graphstorm.git
+
+    cd /path-to-graphstorm/docker/
+
+    bash build_docker_sagemaker.sh ../ <DEVICE_TYPE> <IMAGE_NAME> <IMAGE_TAG>
+
+The ``build_docker_sagemaker.sh`` script takes three arguments:
+
+1. **path-to-graphstorm** (**required**), is the path of the ``graphstorm`` folder, where you cloned the GraphStorm source code. For example, the path could be ``/code/graphstorm``.
+2. **DEVICE_TYPE**, is the intended device type of the to-be built Docker image. Please specify ``cpu`` for building CPU-compatible images for partition job.
```

Comment:
So we do not want users to use `GPU` base images?

Reply:
Yes, we expect the partition job is taken on CPU base image.